src/main/java/com/iota/iri/zmq/MessageQ.java

```diff
@@ -51,7 +51,7 @@ public void shutdown() {
             LOG.error("Publisher service shutdown failed.", e);
         }
 
-        publisher.close();
-        context.term();
+        if(publisher != null) publisher.close();
```

Comment:
Please use brackets and spaces:
```
if (publisher != null) {
    publisher.close();
}
```
This is because #570 that should be merged soon.